### ISSUE TITLE
Expose "launch reason"

### ISSUE THREAD

**Issue by bolsinga** [bugsnag/bugsnag-cocoa#866]
### Description
It would be nice to know "why is the app launching?" Basically we'd like to know if we're re-launching due to a crash, or an OOM, etc. This will allow us to monitor what is going wrong more closely.

**Describe the solution you'd like**
Right now there is `+[Bugsnag appDidCrashLastLaunch]`. How about an enumeration about why the application launched? Something along the lines of https://eng.uber.com/startup-reason-reporter/

**Describe alternatives you've considered**
Right now we use private interfaces in Bugsnag to get `-[BSGOutOfMemoryWatchdog didOOMLastLaunch]` in v5, and `-[BugsnagClient shouldReportOOM]` in v6

Related requests with #570. 

**Additional context**
We basically just want to monitor with our own services if Bugsnag believes an OOM has occurred.

**Comment by xander-jones** [comment]
Hey @bolsinga, thanks for the suggestion! Would you expect this to look like a more nuanced version of `appDidCrashLastLaunch()`? Such that it would return something like: `none`, `unhandledError`, `appUpdated`, `osUpdated`… `oom` rather than just a BOOL?

**Comment by bolsinga** [comment]
Yes, that is what I was thinking. The `-appDidCrashLastLaunch` method could remain for backwards compatibility. The enum could be returned for apps that may want to dig into more of the why. Thanks for your consideration.
